internal/martian/proxy_test.go

```diff
@@ -485,6 +485,77 @@ func TestIntegrationHTTP101SwitchingProtocols(t *testing.T) {
 	}
 }
 
+func TestIntegrationHTTP304NotModified(t *testing.T) {
```

Comment:
thought this test was designed to reproduce the error we encountered with a closed connection when returning a 304 response code. However, it seems the test is already passing on the main branch. Shouldn't it be failing there?